### PR TITLE
Apache camel enable sqs propagation

### LINKED ISSUE #2079: Enable SQS context propagation for apache camel instrumentation
**Is your feature request related to a problem? Please describe.**
Current Apache Camel instrumentation does not support SQS context propagation out of the box. Even though headers are sent as message properties by PRODUCER, values are not deserialised properly by a CONSUMER.

**Describe the solution you'd like**
For SQS routes context propagation should be possible.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by kubawach**
Closes #2079

Makes Camel automatically read message fields as set by current propagators, enabling context propagation.

Remark - in order to avoid muzzle failures for camel users without aws module, aws (SQS) specific code implemented as dynamic method handle invocations.

### CODE REVIEW

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/SqsEndpointModel.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  private final Endpoint endpoint;
+
+  public SqsEndpointModel(Endpoint endpoint) {
+    this.endpoint = endpoint;
+  }
+
+  private Object getSqsConfiguration()
+      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    Method method = endpoint.getClass().getMethod("getConfiguration");
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/SqsEndpointModel.java:22]: Consider using `MethodHandle`s instead, as they're much faster than reflection (see `CouchbaseQueryNormalizer` for examples)

File: instrumentation/apache-camel-2.20/javaagent-unittests/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/MessageAttributeNamesFactoryTest.groovy [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import spock.lang.Specification
+
+class MessageAttributeNamesFactoryTest extends Specification {
+
+  def "should Add Single Field To Empty Names List"() {
+
+    setup:
+    // baggage = single field
+    GlobalOpenTelemetry.set(
+      DefaultOpenTelemetry.builder()
```
Comment (jkwatson) [inline instrumentation/apache-camel-2.20/javaagent-unittests/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/MessageAttributeNamesFactoryTest.groovy:24]: Why use `DefaultOpenTelemetry` up here, and the global in the next test?

Comment (jkwatson) [reply instrumentation/apache-camel-2.20/javaagent-unittests/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/MessageAttributeNamesFactoryTest.groovy:24]: whoops...misread. Is the goal with this to have a propagator-only implementation installed?

Comment (jkwatson) [inline instrumentation/apache-camel-2.20/javaagent-unittests/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/MessageAttributeNamesFactoryTest.groovy:24]: Is the goal with using `DefaultOpenTelemtery` to make sure you have a propagator-only implementation?

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/SqsEndpointModel.java (hunk 1 above)
Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/SqsEndpointModel.java:22]: Done :) Tho in the configuration case I had to use reflection as SqsConfiguration (rtype) can also be unavailable in the runtime.

File: instrumentation/apache-camel-2.20/javaagent-unittests/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/MessageAttributeNamesFactoryTest.groovy (hunk 2 above)
Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent-unittests/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/MessageAttributeNamesFactoryTest.groovy:24]: The goal was rather "it's available in the scope and works" :-) Would it be more kosher in your opinion to import and use the SDK in tests?

Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent-unittests/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/MessageAttributeNamesFactoryTest.groovy]: I eventually removed that class and test ;)

**Comment by kubawach** [comment]
Reworked to accommodate changes done for AWS SDK propagation + new tests :)

**Comment by kubawach** [comment]
Rebased to main (again).

**Review by anuraaga — APPROVED** [review]

File: instrumentation/apache-camel-2.20/javaagent-unittests/apache-camel-2.20-javaagent-unittests.gradle [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apply from: "$rootDir/gradle/java.gradle"
+
+dependencies {
+  testImplementation project(':instrumentation:apache-camel-2.20:javaagent')
+  testImplementation group: 'org.apache.camel', name: 'camel-core', version: '2.20.1'
+  testImplementation group: 'org.apache.camel', name: 'camel-aws', version: '2.20.1'
+
+  testImplementation deps.opentelemetryTraceProps
+  testImplementation deps.bytebuddy
```
[the remark this answers is not in the document]
Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent-unittests/apache-camel-2.20-javaagent-unittests.gradle:9]: Removed.

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by iNikem** [comment]
Trying to trigger

**Pull request merged**